market_maker_keeper/bitso_market_maker_keeper.py

```diff
@@ -0,0 +1,191 @@
+        with Lifecycle() as lifecycle:
+            lifecycle.initial_delay(10)
+            lifecycle.every(1, self.synchronize_orders)
+            lifecycle.on_shutdown(self.shutdown)
+
+    def shutdown(self):
+        self.order_book_manager.cancel_all_orders()
+
+    def pair(self):
+        return self.arguments.pair.lower()
+
+    def token_sell(self) -> str:
+        return self.arguments.pair.split('_')[0].lower()
+
+    def token_buy(self) -> str:
+        return self.arguments.pair.split('_')[1].lower()
+
+    def our_available_balance(self, our_balances: list, token: str) -> Wad:
+        balance = list(filter(lambda x: x['currency'] == token, our_balances))[0]['total']
+        return Wad.from_number(balance)
+
+    def our_sell_orders(self, our_orders: list) -> list:
+        return list(filter(lambda order: order.is_sell, our_orders))
+
+    def our_buy_orders(self, our_orders: list) -> list:
+        return list(filter(lambda order: not order.is_sell, our_orders))
+
+    def synchronize_orders(self):
+        bands = Bands.read(self.bands_config, self.spread_feed, self.control_feed, self.history)
+        order_book = self.order_book_manager.get_order_book()
+        target_price = self.price_feed.get_price()
+
+        # Cancel orders
+        cancellable_orders = bands.cancellable_orders(our_buy_orders=self.our_buy_orders(order_book.orders),
+                                                      our_sell_orders=self.our_sell_orders(order_book.orders),
+                                                      target_price=target_price)
+        if len(cancellable_orders) > 0:
+            self.order_book_manager.cancel_orders(cancellable_orders)
+            return
+
+        # Do not place new orders if order book state is not confirmed
+        if order_book.orders_being_placed or order_book.orders_being_cancelled:
+            self.logger.debug("Order book is in progress, not placing new orders")
+            return
+
+        # Place new orders
+        self.place_orders(bands.new_orders(our_buy_orders=self.our_buy_orders(order_book.orders),
+                                           our_sell_orders=self.our_sell_orders(order_book.orders),
+                                           our_buy_balance=self.our_available_balance(order_book.balances, self.token_buy()),
+                                           our_sell_balance=self.our_available_balance(order_book.balances, self.token_sell()),
+                                           target_price=target_price)[0])
+
+    def place_orders(self, new_orders):
+        def place_order_function(new_order_to_be_placed):
+            amount = new_order_to_be_placed.pay_amount if new_order_to_be_placed.is_sell else new_order_to_be_placed.buy_amount
+            
+            # Convert wad to float as Bitso limits decimal places to 8
+            float_price = Wad.__float__(new_order_to_be_placed.price)
+            float_amount = Wad.__float__(Wad.__round__(amount, 5))
```

Comment:
`Wad` has a _scale_ of 18, so rounding it by 5 gives a _scale_ of 13 before casting to `float`.  How does this rounding relate to Bitso's _precision_ limit of 8?

Reply:
If I were implementing, I'd cast to `float` first and then `round` to 8 decimal places.

Reply:
Thanks for the suggestion, that makes sense. I will update accordingly